tests/ssl-opt.sh

```diff
@@ -5530,7 +5530,7 @@ requires_config_enabled MBEDTLS_SSL_RENEGOTIATION
 requires_config_enabled MBEDTLS_SSL_PROTO_TLS1_2
 run_test    "Renegotiation: gnutls server strict, client-initiated" \
             "$G_SRV --priority=NORMAL:-VERS-ALL:+VERS-TLS1.2:%SAFE_RENEGOTIATION" \
-            "$P_CLI debug_level=3 exchanges=1 renegotiation=1 renegotiate=1" \
+            "$P_CLI force_version=tls12 debug_level=3 exchanges=1 renegotiation=1 renegotiate=1" \
```

Comment:
I am a bit surprise we need that because the GnuTLS server is configured in TLS 1.2 only. In which all.sh component the test is failing without that?

Reply:
Multiple ones (https://jenkins-mbedtls.oss.arm.com/blue/organizations/jenkins/mbed-tls-pr-head/detail/PR-8526-head/34/pipel) but I think they all use the full config with some variation.

Reply:
Do you have an idea or a clue of why enabling MBEDTLS_SSL_RECORD_SIZE_LIMIT in the full configuration does have an effect on this test?

Reply:
I am trying to look into it but in general this is a TLS 1.2 test and we don't support record size extension in TLS 1.2 for now so we weren't supposed to be sending the record size extension in the first place (which we were doing) so this was not a failure in something we were supposed to support but I will try to know the exact reason behind this.

Reply:
This is due to exceeding the number of maximum extensions defined in GnuTLS v3.4.6 which 32 by sending TLS 1.3 handshake with TLS 1.3 extensions.

Reply:
Thanks for getting to the bottom of it. I think the test where the client proposes both TLS 1.2 and TLS 1.3, then TLS 1.2 is selected because the server supports only TLS 1.2 and then the client initiate a renegotiation is a valuable test. Thus rather than forcing TLS 1.2 on client side because of the limitations of G_SRV, what about to use G_NEXT_SRV (version 3.7.2 on the CI) instead for all renegotiation tests involving a GnuTLS server? It would be probably better to add a comment along the "# Tests for renegotiation" explaining why we use G_NEXT_SRV and not G_SRV in Renegotiation tests.